docs/tutorial/building-a-dataset/index.md

```diff
@@ -0,0 +1,145 @@
+We can now move on to implementing some of the QOF business rules.
+
+In this tutorial, we'll look at rule DM\_006, which is about finding the percentage of patients on the register, with a diagnosis of clinical proteinuria or micro-albuminuria who are currently treated with angiotensin-converting enzymes (ACEs) or angiotensin II receptor blockers (ARBs).
+
+The QOF rules define "currently treated" to mean having a medication within 180 days of the index date.
+
+To do this, we'll use these codelists: [`PRT_COD`][1], [`MAL_COD`][2], [`ACE_COD`][3], and [`ARB_COD`][4].
+
+We have already seen how to determine whether a patient has a clinical event matching a code in a codelist.
+Medications are recorded in a similar way to clinical events, and can be queried using the same kinds of ehrQL.
+
+
+```ehrql
+from ehrql import codelist_from_csv, days
+from ehrql.tables.core import clinical_events, medications
+
+
+index_date = "2024-03-31"
+
+proteinuria_codes = codelist_from_csv("codelists/prt_cod.csv", column="code")
+microalbuminuria_codes = codelist_from_csv("codelists/mal_cod.csv", column="code")
+ace_codes = codelist_from_csv("codelists/ace_cod.csv", column="code")
+arb_codes = codelist_from_csv("codelists/arb_cod.csv", column="code")
+
+has_proteinuria_diagnosis = (
```

Comment:
I think it's worth considering whether to use a slightly different pattern here, because I suspect the code here will become a template for lots of future ehrQL.

The pattern I think it might be worth encouraging is to get people to create intermediate frames filtered by date, rather than repeatedly applying data filters.

In this instance those might look like:
```py
previous_events = clinical_events.where(clinical_events.is_before(index_date))
recent_meds = medications.where(medications.date.is_between(index_date - days(180), index_date))